src/databases/Blueprint/avtBlueprintFileFormat.C

```diff
@@ -747,6 +744,88 @@ avtBlueprintFileFormat::AddBlueprintMeshAndFieldMetadata(avtDatabaseMetaData *md
     }
 }
 
+// ****************************************************************************
+// helper method used to add expression meta data for a blueprint mesh.
+//
+// Mark C. Miller, Wed May  6 12:26:33 PDT 2020
+// ****************************************************************************
+static void
+AddBlueprintExpressionMetadata(avtDatabaseMetaData *md, string const &mesh_name,
+    const Node &n_mesh_info)
+{
+    if (!n_mesh_info.has_child("expressions"))
+        return;
+
+    BP_PLUGIN_INFO("adding expressions for " <<  mesh_name);
+
+    NodeConstIterator exprs_itr = n_mesh_info["expressions"].children();
+
+    while (exprs_itr.has_next())
+    {
+        const Node &n_expr = exprs_itr.next();
+        string expname = exprs_itr.name();
+        string exp_topo_name = n_expr["topology"].as_string();
+        string exp_mesh_name = mesh_name + "_" + exp_topo_name;
+        string expname_wmesh = exp_mesh_name + "/" + expname;
+        int ncomps = n_expr["number_of_components"].to_int();
+
+        // Lookup what we already know about this mesh from database metadata
+        const avtMeshMetaData *mmd = md->GetMesh(mesh_name + "_" + exp_topo_name);
+        int ndims = mmd ? mmd->spatialDimension : 0;
+
+        Expression expr;
+        expr.SetName(expname_wmesh);
+        if (ncomps == 1)
+            expr.SetType(Expression::ScalarMeshVar);
+        else if (ndims == 2 && ncomps == 2)
+            expr.SetType(Expression::VectorMeshVar);
+        else if (ndims == 2 && ncomps == 3)
+            expr.SetType(Expression::SymmetricTensorMeshVar);
+        else if (ndims == 2 && ncomps == 4)
+            expr.SetType(Expression::TensorMeshVar);
+        else if (ndims == 3 && ncomps == 3)
+            expr.SetType(Expression::VectorMeshVar);
+        else if (ndims == 3 && ncomps == 6)
+            expr.SetType(Expression::SymmetricTensorMeshVar);
+        else if (ndims == 3 && ncomps == 9)
+            expr.SetType(Expression::TensorMeshVar);
+        else
+            expr.SetType(Expression::ArrayMeshVar);
+
+        // Find all variable references in defn and replace with
+        // parent mesh name. 
+
+        std::string defn = n_expr["definition"].as_string();
+        size_t ltidx=0;
+        while ((ltidx = defn.find('<', ltidx)) != std::string::npos)
+        {
+            size_t gtidx = defn.find('>', ltidx+1);
+            std::string vname(defn, ltidx+1, gtidx-ltidx-1);
+            size_t slash = vname.find('/', 0);
+
+            if (slash)
```

Comment:
Here is logic to NOT do the variable name replacement with mesh-prepending if it appears as though the original variable in the expression already contains it